src/urllib3/__init__.py

```diff
@@ -30,17 +30,19 @@
 except ImportError:
     pass
 else:
-    # fmt: off
-    if (
-        not ssl.OPENSSL_VERSION.startswith("OpenSSL ")
-        or ssl.OPENSSL_VERSION_INFO < (1, 1, 1)
-    ):  # Defensive:
+    if not ssl.OPENSSL_VERSION.startswith("OpenSSL "):  # Defensive:
+        warnings.warn(
+            "urllib3 v2.0 only supports OpenSSL 1.1.1+, currently "
+            f"the 'ssl' module is compiled with {ssl.OPENSSL_VERSION!r}. "
+            "See: https://github.com/urllib3/urllib3/issues/3020",
+            exceptions.NotOpenSSLWarning,
+        )
+    elif ssl.OPENSSL_VERSION_INFO < (1, 1, 1):  # Defensive:
         raise ImportError(
             "urllib3 v2.0 only supports OpenSSL 1.1.1+, currently "
-            f"the 'ssl' module is compiled with {ssl.OPENSSL_VERSION}. "
+            f"the 'ssl' module is compiled with {ssl.OPENSSL_VERSION!r}. "
```

Comment:
A little nitpick ‒ an error message in "Common upgrading issues" needs quotes around "OpenSSL 1.0.2k-fips  26 Jan 2017" after this change 🙂
https://github.com/urllib3/urllib3/blob/de9caa83ed6dc1ad26d78329fa4298131aed46e2/docs/v2-migration-guide.rst?plain=1#L190

Reply:
Thanks, done in [`4e26d7f` (#3024)](https://github.com/urllib3/urllib3/pull/3024/commits/4e26d7f7dd5438379c4bb3fbbde113282a703603)